File: src/main/java/net/alexwells/roomery/proxy/CommonProxy.java
package net.alexwells.roomery.proxy;

import net.alexwells.roomery.Roomery;
import net.alexwells.roomery.block.RoomHolderBlock;
import net.alexwells.roomery.holder.BlockHolder;
import net.alexwells.roomery.item.RoomCardItem;
import net.alexwells.roomery.tile.RoomHolderTileEntity;
import net.minecraft.block.Block;
import net.minecraft.item.Item;
import net.minecraft.item.ItemBlock;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.event.RegistryEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.registry.GameRegistry;

public class CommonProxy {
    @SubscribeEvent
    public void registerBlocks(RegistryEvent.Register<Block> event) {
        event.getRegistry().register(new RoomHolderBlock());
        GameRegistry.registerTileEntity(RoomHolderTileEntity.class, new ResourceLocation(Roomery.MOD_ID, RoomHolderBlock.NAME));
    }

    @SuppressWarnings("ConstantConditions")
    @SubscribeEvent
    public void registerItems(RegistryEvent.Register<Item> event) {
        event.getRegistry().register(new RoomCardItem());

        event.getRegistry().register(new ItemBlock(BlockHolder.roomHolder).setRegistryName(Roomery.MOD_ID, RoomHolderBlock.NAME));
    }
}

File: src/main/java/net/alexwells/roomery/proxy/CommonProxy.kt
package net.alexwells.roomery.proxy

import net.alexwells.roomery.Registry
import net.alexwells.roomery.mechanic.room.world.ROOMERY_DIMENSION_RESOURCE
import net.alexwells.roomery.mechanic.room.world.RoomeryDimensionType
import net.minecraft.block.Block
import net.minecraft.inventory.container.ContainerType
import net.minecraft.item.Item
import net.minecraft.tileentity.TileEntityType
import net.minecraft.world.dimension.DimensionType
import net.minecraftforge.common.DimensionManager
import net.minecraftforge.event.RegistryEvent
import net.minecraftforge.event.world.RegisterDimensionsEvent
import net.minecraftforge.eventbus.api.SubscribeEvent

abstract class CommonProxy {
    @SubscribeEvent
    fun registerBlocks(event: RegistryEvent.Register<Block>) {
        Registry.blocks.forEach { event.registry.register(it) }
    }

    @SubscribeEvent
    fun registerTiles(event: RegistryEvent.Register<TileEntityType<*>>) {
        Registry.tiles.forEach { event.registry.register(it) }
    }

    @SubscribeEvent
    fun registerItems(event: RegistryEvent.Register<Item>) {
        Registry.items.forEach { event.registry.register(it) }
    }

    @SubscribeEvent
    fun registerContainers(event: RegistryEvent.Register<ContainerType<*>>) {
        Registry.containers.forEach { event.registry.register(it) }
    }

    @SubscribeEvent
    fun registerVanillaDimension(event: RegisterDimensionsEvent) {
        if (DimensionType.byName(ROOMERY_DIMENSION_RESOURCE) == null) {
            DimensionManager.registerDimension(ROOMERY_DIMENSION_RESOURCE, RoomeryDimensionType, null, true)
        }
    }
}
